emulator/WScriptShell.js

```diff
@@ -51,20 +51,29 @@ function WScriptShell() {
 		if (process.argv.indexOf("--no-shell-error") === -1)
 			throw new Error("If you can read this, re-run box.js with the --no-shell-error flag.");
 	};
+	this._reg_entries = {
+		"HKLM\\SOFTWARE\\MICROSOFT\\WINDOWS NT\\CURRENTVERSION\\CURRENTVERSION": "5.1",
+		"HKLM\\SOFTWARE\\MICROSOFT\\WINDOWS NT\\CURRENTVERSION\\SYSTEMROOT": "C:\\WINDOWS",
+	};
+	this._normalize_reg_key = (key) => {
+		key = key.toUpperCase().replace("HKEY_LOCAL_MACHINE", "HKLM");
+		key = key.replace("HKEY_CLASSES_ROOT", "HKCR").replace("HKEY_USERS", "HKU");
+		key = key.replace("HKEY_CURRENT_USER", "HKCU").replace("HKEY_CURRENT_CONFIG", "HKCC");
+		return key;
+	};
 	this.regread = (key) => {
-		key = key.toUpperCase();
+		key = this._normalize_reg_key(key);
 		console.log(`Reading registry key ${key}`);
-		switch (key) {
-			case "HKEY_LOCAL_MACHINE\\SOFTWARE\\MICROSOFT\\WINDOWS NT\\CURRENTVERSION\\CURRENTVERSION":
-				return "5.1";
-			case "HKLM\\SOFTWARE\\MICROSOFT\\WINDOWS NT\\CURRENTVERSION\\SYSTEMROOT":
-				return "C:\\WINDOWS";
-			default:
-				console.log("Unknown registry key!");
-				return;
-		}
+		if (!(key in this._reg_entries))
```

Comment:
Suppose that `key = "HKLM\\SOFTWARE\\MICROSOFT\\WINDOWS NT\\CURRENTVERSION\\CURRENTVERSION"`. Then, `key in this._reg_entries` would be true. Then, `!(key in this._reg_entries)` would be false. Then, this statement wouldn't be executed, and the program would report `Unknown registry key!`.